docs/develop/javascript-libraries.md

```diff
@@ -0,0 +1,139 @@
+
+## Array
+
+The [Array](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array) object gives your script functions to work with array types. While arrays are standard JavaScript constructs, they relate to Office Scripts in two major ways: ranges and collections.
+
+Ranges contain several two-dimensional arrays that directly map to the cells in that range. These include properties such as `values`, `formulas`, and `numberFormat`. Your script needs to load the related property before reading any cell in that array, for example `myRange.load("values")` (as well as `await context.sync();` to complete the `load` operation).
+
+The following script searches the **A1:D4** range for any number format containing a "$". The script sets the fill color in those cells to "yellow".
+
+```TypeScript
+async function main(context: Excel.RequestContext) {
+  // Get the range From A1 to D4.
+  let range = context.workbook.worksheets.getActiveWorksheet().getRange("A1:D4");
+
+  // Load the numberFormat property on the range.
+  range.load("numberFormat");
+  await context.sync();
+
+  // Iterate over the entire range.
+  range.numberFormat.forEach((rowItem, rowIndex) => {
+    range.numberFormat[rowIndex].forEach((columnItem, columnIndex) => {
+      // Treat the numberFormat as a string so we can do text comparisons.
+      let columnItemText = columnItem as string;
+      if (columnItemText.indexOf("$") >= 0) {
+        // Set the cell's fill to yellow.
+        range.getCell(rowIndex, columnIndex).format.fill.color = "yellow";
+      }
+    });
+  });
+}
+```
+
+Many Excel objects are contained in a collection. For example, all [Shapes](/javascript/api/office-scripts/excel/excel.shape) in a worksheet are contained in a [ShapeCollection](/javascript/api/office-scripts/excel/excel.shapecollection) (as the `Worksheet.shapes` property). These `*Collection` objects all contain an `items` property, which is an array that stores the objects inside that collection. This can be treated like a normal JavaScript array, but the items in the collection have to first be loaded. If you need to work with a property on every object in the collection, use a hierarchal load statement (`items/propertyName`).
+
+The following script logs the type of every shape in the current worksheet.
+
+```TypeScript
+async function main(context: Excel.RequestContext) {
+  // Get the current worksheet.
+  let selectedSheet = context.workbook.worksheets.getActiveWorksheet();
+
+  // Get the shapes in this worksheet.
+  let shapes = selectedSheet.shapes;
+  shapes.load("items/type");
+  await context.sync();
+
+  // Log the type of every shape.
```

Comment:
Consider updating this comment to convey its relevance to collections. ->

// Log the type of every shape in the collection.